### PR TITLE
77 game state

### PR DESCRIPTION
**Pull request by emmettirl**

### CODE REVIEW

File: scripts/battle/GameState.cs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        if (PlayerHealth > MaxPlayerHealth)
+        {
+            PlayerHealth = MaxPlayerHealth;
+        }
+        
+        if (!IsPlayerAlive())
+        {
+            //end round on loss
+        }
+    }
```
Comment (VonJellybean) [inline scripts/battle/GameState.cs:60]: Like when we discussed negative damage vs healing when it comes to enemies, we'll need to separate this out.

File: scripts/battle/GameState.cs [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+    // Combo Methods
+
+    public void ProcessCardPlayed(Card card) //largely based on Cath's python code
```
Comment (VonJellybean) [inline scripts/battle/GameState.cs:64]: I don't like this name, Cards process their own playing so I don't want another method named like this.

File: scripts/battle/GameState.cs [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        int multiplierAdjustment = comboValue - blunderValue;
+        
+        //adjust combo multiplier
+        ComboMultiplier += multiplierAdjustment;
+        if (ComboMultiplier < 1) { ComboMultiplier = 1; }
+        
+        //resolve other combo effects
+    }
+    
+    public void ProcessCombo(int number) // overloaded to allow int input for end round
```
Comment (VonJellybean) [inline scripts/battle/GameState.cs:128]: You have a null safety check in the above method that duplicates this behaviour.

Or, leave this method in place and have the above method just call  `ProcessCombo(matchingCombo.CardList.Count)`

**Review by VonJellybean — APPROVED** [review]
Made a number of comments, but I don't think they're blocking ones. I'd like to make a pass over this myself to simplify some things and try and make it a bit more readable.

File: scripts/battle/GameState.cs (hunk 2 above)
Comment (emmettirl) [reply scripts/battle/GameState.cs:64]: Changed this to GamestateCardPlayed()

File: scripts/battle/GameState.cs (hunk 1 above)
Comment (emmettirl) [reply scripts/battle/GameState.cs:60]: Robbed some of your enemy code and did this

**Pull request merged**